src/user/login/login.controller.ts

```diff
@@ -37,7 +37,7 @@ class LoginController {
   ) {
     'ngInject';
     this.userCreationEnabled = Constants.userCreationEnabled;
-    this.localLoginEnabled = Constants.localLoginEnabled;
+    this.localLoginEnabled = Constants.authentication.localLoginEnabled;
```

Comment:
What if `Constants.authentication` is undefined  (no authentication config in constants.json) ?

Reply:
fixed